build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -275,19 +276,50 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
             }
             else
             {
-                ExecIn(cmakeOut, Path.Combine(c.BuildContext.BuildDirectory, "src", "corehost", "build.sh"),
-                        "--arch",
-                        "x64",
-                        "--hostver",
-                        hostVersion.LatestHostVersion.ToString(),
-                        "--fxrver",
-                        hostVersion.LatestHostFxrVersion.ToString(),
-                        "--policyver",
-                        hostVersion.LatestHostPolicyVersion.ToString(),
-                        "--rid",
-                        rid,
-                        "--commithash",
-                        commitHash);
+                string arch;
+                switch (platform.ToLower())
+                {
+                    case "x64":
+                        arch = "x64";
+                        break;
+                    case "arm":
+                        arch = "arm";
+                        break;
+                    default:
+                        throw new PlatformNotSupportedException("Target Architecture: " + platform + " is not currently supported.");
+                }
+
+                if (crossCompiler == null) {
+                    ExecIn(cmakeOut, Path.Combine(c.BuildContext.BuildDirectory, "src", "corehost", "build.sh"),
+                            "--arch",
+                            arch,
+                            "--hostver",
+                            hostVersion.LatestHostVersion.ToString(),
+                            "--fxrver",
+                            hostVersion.LatestHostFxrVersion.ToString(),
+                            "--policyver",
+                            hostVersion.LatestHostPolicyVersion.ToString(),
+                            "--rid",
+                            rid,
+                            "--commithash",
+                            commitHash);
+                } else {
+                    ExecIn(cmakeOut, Path.Combine(c.BuildContext.BuildDirectory, "src", "corehost", "build.sh"),
+                            "--arch",
+                            arch,
+                            "--hostver",
+                            hostVersion.LatestHostVersion.ToString(),
+                            "--fxrver",
+                            hostVersion.LatestHostFxrVersion.ToString(),
+                            "--policyver",
+                            hostVersion.LatestHostPolicyVersion.ToString(),
+                            "--rid",
+                            rid,
+                            "--commithash",
+                            commitHash,
+                            "--xcompiler",
```

Comment:
Can you please refactor this using an ArgList so that we dont duplicate the implementation? Consider using https://github.com/dotnet/core-setup/blob/eca152525e19a41d489cd5845afa14982653cc0a/build_projects/Microsoft.DotNet.Cli.Build.Framework/BuildHelpers.cs#L10